packages/nimble-components/src/wafer-map/specs/README.md

```diff
@@ -0,0 +1,235 @@
+
+-   As an initial step, we submit a skeleton of the `nimble-wafer-map` component into the main branch. This will serve as a starting point on which the complete functionality will be built on.
+-   Once the skeleton is in place, we start to work on and submit individual pieces of the functionality so it's not very overwhelming to review the code.
+
+### States
+
+We are not planning to allow the user to generate an “empty" wafer map, so we must ensure that data is provided when instantiating a new `nimble-wafer-map` component.
+The only action which we could consider as a "state transition" is the zoom functionality which was specified in the sections above.
+We expect that the end user be able to zoom in on the dies within the wafer map and be able to zoom out by using the mouse wheel.
+Also, the user must be allowed to pan the zoomed in wafer map into any direction of its XY axis by holding the left mouse button and moving the mouse cursor around.
+
+Another interaction, which must be part of the component is the ability to identify the location of the mouse pointer on the wafer map die area.
+
+Furthermore `disabled` and `focused` state will also be supported by the component.
+
+### Accessibility
+
+Important: this is out of scope for the current implementation.\
+Since the `nimble-wafer-map` component was designed to display tens of thousands of data points, it's challenging to provide Accessibility functionality which can read out loud the values displayed in a meaningful manner.
+Based on the WAI documentation on [Complex Images](https://www.w3.org/WAI/tutorials/images/complex/), in the future we could provide a paragraph containing information about the name of the wafer that is displayed, the LOT where the wafer comes from and a general overview (e.g. % of the faulty dies on the wafer) of the status. This might give a pretty good description about what the displayed nimble component shows to the user.
+Please note that in the initial implementation we do not plan to gather and summarize any of this information so this will be part of future updates to this component.
+Currently the only way to navigate and interact with the wafer map is only by using the mouse. Future updates might include functionality which allows the user to tab into the wafer map area and move around in the die area using the keyboard arrows.
+
+### Globalization
+
+Currently there will be no string data presented by this component. Might change in the future, but for now it will contain only numerical data. Similar to the `nimble-number-field`, initially we don't plan to support locale-specific numeric formatting.
+
+### Security
+
+N/A
+
+### Performance
+
+As mentioned in the "Risks" section, the biggest challenge from the performance standpoint is the representation of large number of datapoints and interacting with these datapoints in the browser.
```

Comment:
It sounds like this is important and mentioned a couple times that lots of data may be passed to the component. Do you have expected performance requirements?

I would be interested in:
- Expected data set size max ranges, up to 1K, 10K, 100K?
- Expected update time? Data refreshes every 10s, 1s, 100ms?
- How long after setting to render? Draw is within 10ms, 100ms, 1s, 3s?

Reply:
For the initial release we do not have any performance expectation from the component. The only requirement for now is to provide empirical data on render time for: single wafer with e.g. 50k data points, multiple wafers with 50k data points, how increasing the number of wafers affects the rendering time etc. We would like to do this benchmarking once we are done with the implementation.

Reply:
> For the initial release we do not have any performance expectation from the component.

I'm sure there are some expectations. If it took 5 seconds to render 10 data points I doubt that would be acceptable. 😋
If we have some basic lower bounds based on expected data set sizes that would be great to capture in the spec.

> The only requirement for now is to provide empirical data on render time for: single wafer with e.g. 50k data points, multiple wafers with 50k data points, how increasing the number of wafers affects the rendering time etc. We would like to do this benchmarking once we are done with the implementation.

If those are the initial benchmarks that sounds worth capturing in the spec and the spec can be updated as we refine the performance goals 👍

Reply:
@rajsite yeah 10 data points in 10 seconds would be a disaster. I'll set some goals for the benchmarking, something like: 
100 data points
1k data points
10k data points
100k data points
 
just to have a reference point.

Reply:
Updated the performance requirements with a few estimates. However, these are not based on real world data, they set a pretty clear benchmark for future optimizations.